tests/unit/lib/foxbox/sequential-timer_test.js

```diff
@@ -0,0 +1,126 @@
+import { waitForNextMacroTask } from '../../test-utils';
```

Comment:
Repeat myself - testing promise based code when promises are not exposed via public api is really __hard__ :/ If browser supports latest Micro and Macro task specs (like browsers we support) we can rely on ```waitForNextMacroTask```